kratos/geometries/geometry.h

```diff
@@ -4102,6 +4083,26 @@ class Geometry
     ///@name Protected Inquiry
     ///@{
 
+    /**
+     * @brief Checks if the geometry points are valid
+     * Checks if the geometry points are valid from the pointer value
+     * Points are not valid when the pointer value is null
+     * @return true All points are valid
+     * @return false At least one point has nullptr value
+     */
+    bool AllPointsAreValid() const
+    {
+        // Check if the geometry has valid (i.e. not nullptr) points
+        bool is_valid = true;
+        for (IndexType i = 0; i < this->PointsNumber(); ++i) {
+            if (this->pGetPoint(i) == nullptr) {
+                is_valid = false;
+                break;
+            }
+        }
+        return is_valid;
```

Comment:
or maybe even better:

```suggestion
        return std::none_of(mPoints.begin(), mPoints.end(), [](auto& pPoint) { return pPoint == nullptr; }))
```

Reply:
Cool. I'll do it like this. Thanks for the hint.

Reply:
Done.